docs/05-rfcs/2023-01-21-wing-plugins.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Wing Plugins
+description: How to implement wing plugins
+---
+
+# Wing Plugin System
+
+- **Author(s)**: @Hasanaburayyan
+- **Submission Date**: {2023-01-21}
+- **Stage**: Draft
+
+Implementing a simple wing plugin system
+
+## Background
+
+Developers using wing should whenever possible only care about functional aspects of an application's infrastructure. However, there must be a way to define non-functional requirements as that will be crucial for wide adoption. For instance, an organization may have specific security requirements to adhere to for compliance and regulation sake, such as encryption at rest, versioning, replication etc... 
+
+Applying these non-functional requirements will mean that the wing compiler should provide the ability for users to write plugins that can hook into the compilation process and apply business logic to resources.
+
+## Proposal
+
+Initially the plugin system can be crafted in simple manner, allowing the ability for users to pass in additional `--plugins` option to the wing cli. These plugin files will be Javascript that contains steps for applying non-functional infrastructure requirements.
```

Comment:
Mention that people can pass multiple plugins and that they will be executed in order (for each hook).